File: backend/src/main/java/com/github/liliancharlotte/model/JobPosting.java
package com.github.liliancharlotte.model;

import org.springframework.data.annotation.Id;

public record JobPosting(
        @Id
        String id,
        String companyName,
        boolean isUnsolicited,
        String jobTitle,
        String jobDescription,
        String jobPostingLink,
        WorkModel remote,
        String locatedAt,
        ColumnStatus status) {
}

File: frontend/src/model/JobPosting.ts
export type ColumnStatus =
    "INTERESTED_IN"
    | "CURRENTLY_WORKING_ON"
    | "APPLICATION_SUBMITTED"
    | "INTERVIEW_SCHEDULED"
    | "DECIDING_ON_JOB_OFFERING"
export type WorkModel = "IN_OFFICE" | "REMOTE" | "HYBRID"

export type JobPosting = {
    id: string,
    companyName: string,
    isUnsolicited: boolean,
    jobTitle: string,
    jobDescription: string,
    jobPostingLink: string,
    remote: WorkModel,
    locatedAt: string,
    status: ColumnStatus,
    applicationSubmissionDate: string
}

export type JobPostingRequest = {
    companyName: string,
    isUnsolicited: boolean,
    jobTitle: string,
    jobDescription: string,
    jobPostingLink: string,
    remote: WorkModel,
    locatedAt: string,
    status: ColumnStatus,
    applicationSubmissionDate: string
}
